Language suggestions and restructure of #1022 (#1071)
codenvy/che-docs: src/main/pages/che-7/end-user-guide/assembly_che-theia-remote-plugin-image.adoc

// This assembly is included in the following assemblies:
//

> // assembly_what-is-a-che-theia-plug-in.adoc

Reviewer: ```suggestion
// assembly_what-is-a-che-theia-plug-in.adoc

:parent-context-of-remote-plug-in-endpoint: {context}
```

Reviewer: Plus
```
:context: {parent-context-of-remote-plug-in-endpoint}
```
should be added at the end of the file.

[id="remote-plug-in-endpoint_{context}"]
= Remote plug-in endpoint

:context: remote-plug-in-endpoint

[NOTE]
====
This document applies only to {prod} 7.3.1 and later.
====

> {prod} has a remote plug-in endpoint service to start VS Code Extensions and Che-Theia plug-ins in the separated containers. {prod} injects the remote plug-in endpoint binaries into each remote plug-in container. This service starts your remote extensions and plug-ins defined in the plug-in `meta.yaml` file and connects them to the Che-Theia editor container.

Reviewer: ```suggestion
{prod} has a remote plug-in endpoint service to start VS Code Extensions and Che-Theia plug-ins in separate containers. {prod} injects the remote plug-in endpoint binaries into each remote plug-in container. This service starts remote extensions and plug-ins defined in the plug-in `meta.yaml` file and connects them to the Che-Theia editor container.
```

> A remote plug-in endpoint creates a plug-in API proxy between your remote plug-in container and Che-Theia editor container. Remote plug-in endpoint is also an interceptor for some plug-in API parts, it launches them inside a remote sidecar container, instead of an editor container. Examples: terminal API, debug API.

Reviewer: ```suggestion
The remote plug-in endpoint creates a plug-in API proxy between the remote plug-in container and the Che-Theia editor container. The remote plug-in endpoint is also an interceptor for some plug-in API parts, which it launches inside a remote sidecar container instead of an editor container. Examples: terminal API, debug API.
```

> A remote plug-in endpoint executable command is stored in the environment variable of the remote plug-in container -`PLUGIN_REMOTE_ENDPOINT_EXECUTABLE`

Reviewer: ```suggestion
The remote plug-in endpoint executable command is stored in the environment variable of the remote plug-in container: `PLUGIN_REMOTE_ENDPOINT_EXECUTABLE`.
```

> {prod} provides two ways to start remote plug-in endpoint with your sidecar image:

Reviewer: ```suggestion
{prod} provides two ways to start the remote plug-in endpoint with a sidecar image:
```

> * defining a launch remote plug-in endpoint using Dockerfile. You must patch an original image and rebuid it.

Reviewer: ```suggestion
* Defining a launch remote plug-in endpoint using a Dockerfile. To use this method, patch an original image and rebuild it.
```

> * defining a launch remote plug-in endpoint in the plug-in a `meta.yaml` file. Use this method if you do not want to patch an original image to start a remote plug-in endpoint.

Reviewer: ```suggestion
* Defining a launch remote plug-in endpoint in the plug-in `meta.yaml` file. Use this method to avoid patching an original image.
```

> include::proc_defining-a-launch-remote-plug-in-endpoint-using-dockerfile.adoc[leveloffset=1]

Reviewer: ```suggestion
include::proc_defining-a-launch-remote-plug-in-endpoint-using-dockerfile.adoc[leveloffset=+1]

```

> include::proc_defining-a-launch-remote-plug-in-endpoint-in-a-meta-yaml-file.adoc[leveloffset=1]

Reviewer: ```suggestion
include::proc_defining-a-launch-remote-plug-in-endpoint-in-a-meta-yaml-file.adoc[leveloffset=+1]
```
